README.md

```diff
@@ -69,30 +69,25 @@ Create an executable file in the .git/hooks directory of the target repository:
 
 ```bash
 #!/bin/bash
-if [ -d ".github/workflows" ]; then
-  if command -v action-validator >/dev/null 2>&1; then
-    echo "Running pre-commit hook for GitHub Actions: https://github.com/mpalmer/action-validator"
-    scan_count=0
-    for action in $(find .github/workflows -name "*.y*ml"); do
-      validate="$(action-validator "$action")"
-      if [ -z "$validate" ]; then
-        echo "✅ $action"
-      else
-        echo "❌ $action"
-        echo "$validate"
-        exit 1
-      fi
-      scan_count=$((scan_count+1))
-    done
-    echo "action-validator scanned $scan_count GitHub Actions found no errors!"
+if ! command -v action-validator >/dev/null; then
+  echo "action-validator is not installed."
+  echo "Installation instructions: https://github.com/mpalmer/action-validator"
+  exit 1
+fi
+echo "Running pre-commit hook for GitHub Actions: https://github.com/mpalmer/action-validator"
+scan_count=0
+for action in $(git diff --cached --name-only --diff-filter=ACM | grep -E "\.github/workflows|actions" | grep -E "\.ya*ml$"); do
```

Comment:
You could combine those two `grep` calls into one, with something like (off the top of my head):

```
grep -E '^\.github/(workflows|actions)/.*\.ya?ml$'
```

(I also anchored the regex at the front, so it avoids a subdir named `.github`, and tightened up the extension match a bit so it won't match `.yaaaaaaaaaaaaaaaaaaaaaml`).

Reply:
nice one. updated.